### ISSUE TITLE
unimim is still unavailable after one year

### ISSUE THREAD

**Issue by jsrdcht** [microsoft/unilm#1579]
A year later, unimim still has not released the source code. A year ago, I spent a long time trying to replicate and analyze this paper, but was unable to do so. Are there any plans to make the source code of this paper available?

To be honest, I am a bit disappointed.

**Comment by pengzhiliang** [comment]
Thanks for your reminder. I will work on it as soon as possible.

**Comment by pengzhiliang** [comment]
In fact, maskdistill is based on beit-2, but the pretraining/finetuing hyper-params are somewhat different. Can you describe the gap/some details/problem when you  replicate it?

**Comment by drkostas** [comment]
Should we expect the code to be released soon, or assume it will stay private?

**Comment by drkostas** [comment]
**tl;dr** I implemented MaskDistill from scratch in PyTorch, reproduced the paper's experiments, and open sourced the code and weights: https://github.com/drkostas/MaskDistill-PyTorch

I needed this for my research and the existing unofficial implementation (https://github.com/bwconrad/masked-distillation) was not giving me the expected results, so I ended up writing everything from scratch. ViT-Base pretrained for 300 epochs with CLIP teacher reproduces:

- Finetuning: 84.8% top-1 (paper reports 85.3%)
- Linear probe: 76.3% top-1
- Semantic segmentation: 52.6 mIoU (paper reports 53.8)
- k-NN: 75.6%
- Object detection: 44.4 bbox mAP
- Instance segmentation: 40.1 segm mAP

All checkpoints on HuggingFace: https://huggingface.co/drkostas/MaskDistill-ViT-Base
